### DISCUSSION TITLE
What is a Tedit "stylesheet" ?

### DISCUSSION THREAD

**Discussion by rmkaplan** [Interlisp/medley#2229] [General]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by rmkaplan** [comment]
Oops, returned to soon.

Here is my issue:  There is a certain amount of complexity in the Tedit code that deals with "style sheets".  But I can't find any mention of them in the Tedit documentation--how are they set, what do they do? 

So, in cleaning this up, it isn't clear what if any of this needs to be preserved--or whether it needs to be enhanced in some user-accessible way.

Does anybody know?

**Comment by masinter** [comment]
I'm not sure of the order of things, but I think "style sheets" --as a way of separating content from its "looks" -- may have been nascent when the bulk of TEdit was built.  TEdit did separate the stream of text from the looks (font, size, etc.) which I think was done for Bravo and BravoX. Perhaps a work in progress? Are there dates associated with the style sheet notion and other TEdit changes?

**Comment by rmkaplan** [comment]
All related functions are from July 1993.

Each piece has its own looks, but it may be that the intent was for some of the fields of a piece to be left out and defaulted to whatever the current style is.  That would be reasonable—to be able to have a “style” for function names that picks a font, size, italics, whatever, and you would just say that a given string in a document is a function.

But there doesn’t seem to be any public interface for this—how do you set up a style sheet?  How do you pick one for a given selection?  For example, there doesn’t seem to be a style item in any of the menus

And, there doesn’t seem to be any code that would address some obvious situations.  What if the definition of a style changes while you are editing a document—should all the affected pieces be updated and reformatted?  Should the styles be saved with the document, so it always starts out the way it was saved? Etc.

**Comment by pamoroso** [comment]
This question may be converted to a discussion.

**Comment by pamoroso** [comment]
Can this issue be moved to a discussion?

**Comment by rmkaplan** [comment]
Yes
